lib/matplotlib/colors.py

```diff
@@ -1226,6 +1256,864 @@ def reversed(self, name=None):
+            return False
+        if len(self) != len(other):
+            return False
+        for c0, c1 in zip(self, other):
+            if c0 != c1:
+                return False
+        if not all(self._rgba_bad == other._rgba_bad):
+            return False
+        if self.combination_mode != other.combination_mode:
+            return False
+        return True
+
+    def __getitem__(self, item):
+        return self._colormaps[item]
+
+    def __iter__(self):
+        for c in self._colormaps:
+            yield c
+
+    def __len__(self):
+        return len(self._colormaps)
+
+    def __str__(self):
+        return self.name
+
+    def get_bad(self):
+        """Get the color for masked values."""
+        return np.array(self._rgba_bad)
+
+    def resampled(self, lutshape):
+        """
+        Return a new colormap with *lutshape* entries.
+
+        Parameters
+        ----------
+        lutshape : tuple of (`int`, `None`)
+            The tuple must have a length matching the number of variates.
+            For each element in the tuple, if `int`, the corresponding colorbar
+            is resampled, if `None`, the corresponding colorbar is not resampled.
+
+        Returns
+        -------
+        MultivarColormap
+        """
+
+        if not np.iterable(lutshape) or len(lutshape) != len(self):
+            raise ValueError(f"lutshape must be of length {len(self)}")
+        new_cmap = self.copy()
+        for i, s in enumerate(lutshape):
+            if s is not None:
+                new_cmap._colormaps[i] = self[i].resampled(s)
+        return new_cmap
+
+    def with_extremes(self, *, bad=None, under=None, over=None):
+        """
+        Return a copy of the `MultivarColormap` with modified out-of-range attributes.
+
+        The *bad* keyword modifies the copied `MultivarColormap` while *under* and
+        *over* modifies the attributes of the copied component colormaps.
+        Note that *under* and *over* colors are subject to the mixing rules determined
+        by the *combination_mode*.
+
+        Parameters
+        ----------
+        bad : None or :mpltype:`color`
+            If Matplotlib color, the bad value is set accordingly in the copy
+
+        under : None or tuple of :mpltype:`color`
+            If tuple, the `under` value of each component is set with the values
+            from the tuple.
+
+        over : None or tuple of :mpltype:`color`
+            If tuple, the `over` value of each component is set with the values
+            from the tuple.
+
+        Returns
+        -------
+        MultivarColormap
+            copy of self with attributes set
+
+        """
+        new_cm = self.copy()
+        if bad is not None:
+            new_cm._rgba_bad = to_rgba(bad)
+        if under is not None:
+            if not np.iterable(under) or len(under) != len(new_cm):
+                raise ValueError("*under* must contain a color for each scalar colormap"
```

Comment:
For multivariate you can do `.with_extremes(over=(None, 'r', None))`, which only sets the `over` color on one of the component colorbars.
For bivariate, you cannot, simply because there are eight different extremes (4 sides + 4 corners), and IMO that is too many for it to make sense to set different colors for all of them, i.e. there is only `outside`.
(IMO in this case it is better to plot the different channels separately, and look at over/under that way)